src/Microsoft.Sbom.Targets/Microsoft.Sbom.Targets.targets

```diff
@@ -0,0 +1,80 @@
+    <SbomGenerationDeleteManifestDirIfPresent Condition=" '$(SbomGenerationDeleteManifestDirIfPresent)' == '' ">true</SbomGenerationDeleteManifestDirIfPresent>
+    <UnzipGuid>$([System.Guid]::NewGuid())</UnzipGuid>
+    <ShortUnzipGuidFolder>$([System.String]::Copy('$(UnzipGuid)').Substring(0, 8))</ShortUnzipGuidFolder>
+  </PropertyGroup>
+
+  <!-- After the Nuget Package is generated, we will unzip, scan, generate the SBOM and zip again. -->
+  <Target Name="GenerateSbomTarget" AfterTargets="Pack" Condition=" '$(GenerateSBOM)' ==  'true'" >
+    <!-- Unzip Nuget package, so it can be scanned by the SBOM Task. -->
+    <PropertyGroup>
+      <PackageOutputFullPath>$([System.IO.Path]::GetFullPath('$(PackageOutputPath)'))</PackageOutputFullPath>
+
+      <NugetPackage>
+        $([System.IO.Path]::Combine($(PackageOutputFullPath), $(PackageId).$(PackageVersion).nupkg))
+      </NugetPackage>
+      <NugetPackageUnzip>
+        $([System.IO.Path]::Combine($(PackageOutputFullPath), $(PackageId).$(PackageVersion).$(ShortUnzipGuidFolder).temp))
+      </NugetPackageUnzip>
+    </PropertyGroup>
+    <Unzip DestinationFolder="$(NugetPackageUnzip)" SourceFiles="$(NugetPackage)" OverwriteReadOnlyFiles="true" ContinueOnError="ErrorAndContinue" />
+
+    <!--
+      Call the SBOM task to generate a SBOM. The SBOM will be generated at the BuildDropPath location, which is the root folder
+      of the NuGet package. NOTE: Currently, the ManifestDirPath is an optional parameter that will not be supported, as the desired location of the SBOM will be
+      at the BuildDropPath directory. However, this parameter is available for future revision and implementation.
+    -->
+    <GenerateSbom
+        BuildDropPath="$(NugetPackageUnzip)"
+        BuildComponentPath="$(SbomGenerationBuildComponentPath)"
+        PackageSupplier="$(SbomGenerationPackageSupplier)"
+        PackageName="$(SbomGenerationPackageName)"
+        PackageVersion="$(SbomGenerationPackageVersion)"
+        NamespaceBaseUri="$(SbomGenerationNamespaceBaseUri)"
+        NamespaceUriUniquePart="$(SbomGenerationNamespaceUriUniquePart)"
+        ExternalDocumentListFile="$(SbomGenerationExternalDocumentReferenceListFile)"
+        FetchLicenseInformation="$(SbomGenerationFetchLicenseInformation)"
+        EnablePackageMetadataParsing="$(SbomGenerationEnablePackageMetadataParsing)"
+        Verbosity="$(SbomGenerationVerbosity)"
+        ManifestInfo="$(SbomGenerationManifestInfo)"
+        DeleteManifestDirIfPresent="$(SbomGenerationDeleteManifestDirIfPresent)"
+        SbomToolPath="$(SbomToolPath)"
+        ContinueOnError="ErrorAndContinue">
+    </GenerateSbom>
+    <Message Importance="High" Text="Task result: $(SbomPathResult)" />
+
+    <!-- Zip the Nuget package back up and delete the temporary unzipped package. -->
+    <ZipDirectory SourceDirectory="$(NugetPackageUnzip)" DestinationFile="$(NugetPackage)" Overwrite="true" ContinueOnError="ErrorAndContinue" />
```

Comment:
On Windows, will this cause the zip's central directories to have forward or back slaches? 

NuGet's C# code that zips normalizes on `/`, and our extraction code expects this and does **NOT** fixup any `\` to `/` on Linux/Mac. This causes NuGet to extract to paths that don't work, and the package doesn't work on Mac/Linux, only on Windows.

Reply:
@dtivel does package signing also require the central directory to be sorted?

Does signing re-sort the central directory if it's not already sorted, so it doesn't matter when unsigned?

Reply:
Interesting comment @zivkan. How can I confirm which slashes are being used in the central directory?

Reply:
Thank you for the comment @zikvan.

Using [this script](https://gist.github.com/gustavoaca1997/a26e49db07dca58aa147fe95ed2a36e3), I confirmed that the package generated has forward slashes:
![image](https://github.com/user-attachments/assets/fd4da66d-911f-493f-9edd-eb6e220af2cb)

Reply:
I also created this PR on NuGet. If it makes it into .NET 9.0.100, hopefully it'll be a non-issue in any case

* https://github.com/NuGet/NuGet.Client/pull/5989